patroni/config.py

```diff
@@ -614,3 +615,25 @@ def get_global_config(self, cluster: Union[Cluster, None]) -> GlobalConfig:
         :returns: :class:`GlobalConfig` object
         """
         return get_global_config(cluster, self._dynamic_configuration)
+
+    def _validate_failover_tags(self):
+        """Checks nofailover/failover_priority config and warns user if it's contradictory"""
```

Comment:
```suggestion
        """Check ``nofailover``/``failover_priority`` config and warn user if it's contradictory."""
```

Reply:
How: It just looks at the tags and gets their value.

> What is contradictory?

When implementing this feature, I adopted the following convention: `failover_priority` <= 0 means that it should _never_ seek leadership. On the flip side, `failover_priority > 0` means leadership is theoretically possible for this node. This made sense to me and was similar to other such priority systems I had seen for this kind of thing (notable MongoDB replication sets).

To preserve sanity (and backwards compatibility) the `nofailover` tag will still exist. A contradictory configuration is one where `nofailover` is `true` but `failover_priority > 0`, or where `nofailover` is `false` but `failover_priority <= 0`. Essentially, `nofailover` and `failover_priority` are communicating different things.

This checks for this edge case (which is a misconfiguration on the part of the user) and warns them. The behaviour is as if `failover_priority` were not provided (i.e `nofailover` is the bedrock source of truth)

Reply:
Cool, thanks for the explanation. The question was more to prompt for what a reader of the docstring might think, it would be worth adding some of what you describe above to the docstring.

So, something like:

```suggestion
        """Compare ``nofailover``/``failover_priority`` tags and warns user of conflicting configuration.
        
        .. note::
          To preserve sanity (and backwards compatibility) the ``nofailover`` tag will still exist. A contradictory
          configuration is one where ``nofailover`` is ``True`` but ``failover_priority > 0``, or where 
          ``nofailover`` is ``False``, but ``failover_priority <= 0``. Essentially, ``nofailover`` and 
          ``failover_priority`` are communicating different things.

          This checks for this edge case (which is a misconfiguration on the part of the user) and warns them. 
          The behaviour is as if ``failover_priority`` were not provided (i.e ``nofailover`` is the
          bedrock source of truth)
        
        """
```

(You might want to check line length before applying)